### PR TITLE
feat(meilisearch): Meilisearch destination plugin

### PR DESCRIPTION
**Pull request by candiduslynx**
Implements #8857

### CODE REVIEW

File: plugins/destination/meilisearch/client/client.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+func New(_ context.Context, logger zerolog.Logger, dstSpec specs.Destination) (destination.Client, error) {
+	var pkColumn string
+	switch dstSpec.WriteMode {
+	case specs.WriteModeAppend:
+		pkColumn = schema.CqIDColumn.Name
+	case specs.WriteModeOverwrite:
+		pkColumn = hashColumnName
+	case specs.WriteModeOverwriteDeleteStale:
```
Comment (disq) [inline plugins/destination/meilisearch/client/client.go:69]:
```suggestion
	default:
```

File: plugins/destination/meilisearch/client/client.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+		return nil, fmt.Errorf("%q write_mode is not supported", specs.WriteModeOverwriteDeleteStale)
```
Comment (disq) [inline plugins/destination/meilisearch/client/client.go:70]:
```suggestion
		return nil, fmt.Errorf("%q write_mode is not supported", dstSpec.WriteMode)
```

File: plugins/destination/meilisearch/client/write.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/cloudquery/plugin-sdk/specs"
+)
+
+func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, data [][]any) error {
+	index, err := c.Meilisearch.GetIndex(table.Name)
+	if err != nil {
+		return err
+	}
+
+	var toDoc func([]any) map[string]any
```
Comment (disq) [inline plugins/destination/meilisearch/client/write.go:17]: Maybe define a type func for clarity?

File: plugins/destination/meilisearch/main.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	sentryDSN = "https://3218256ba1fb4d2fa08eb03feea5e871@o1396617.ingest.sentry.io/4504893018996736"
+)
+
+func main() {
+	serve.Destination(
+		destination.NewPlugin(
+			"meilisearch",
+			plugin.Version,
+			client.New,
+			// destination.WithDefaultBatchSize(10000),
```
Comment (disq) [inline plugins/destination/meilisearch/main.go:20]: Comments left here

**Review by disq — COMMENTED** [review]
few nits

**Comment by candiduslynx** [comment]
We need to perform an additional `GetIndex` call in `Client.getIndexSchema` until  https://github.com/meilisearch/meilisearch-go/pull/426 is merged & released

File: website/pages/docs/plugins/destinations/meilisearch/_configuration.mdx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  name:       meilisearch
+  path:       cloudquery/meilisearch
+  version:    "VERSION_DESTINATION_MEILISEARCH"
+  write_mode: "overwrite"
+  # batch_size: 1000 # optional
+  # batch_size_bytes: 5242880 # optional
+  spec:
+    # meilisearch plugin spec
+    api_key: "<YOUR_MEILISEARCH_API_KEY>"
+    host:    "<YOUR_MEILISEARCH_HOST_PORT>"
```
Comment (hermanschaaf) [inline website/pages/docs/plugins/destinations/meilisearch/_configuration.mdx:15]:
```suggestion
    host:    "localhost:7700"
```

Comment (hermanschaaf) [reply website/pages/docs/plugins/destinations/meilisearch/_configuration.mdx:15]: Or `http://localhost:7700`? Not sure, the docs lower down suggest `http://`

Comment (hermanschaaf) [reply website/pages/docs/plugins/destinations/meilisearch/_configuration.mdx:15]: Yeah looks like it should be `http://localhost:7700`

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Nice, looks good! I haven't tested it yet, but read through the code and docs and left a few comments / questions.

File: plugins/destination/meilisearch/client/client.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+var _ destination.Client = (*Client)(nil)
+
+func (c *Client) Close(context.Context) error {
+	c.Meilisearch = nil
+	return nil
+}
+
+func (*Client) DeleteStale(context.Context, schema.Tables, string, time.Time) error {
+	return fmt.Errorf("DeleteStale not supported")
```
[the remark this answers is not in the document]
Comment (candiduslynx) [reply plugins/destination/meilisearch/client/client.go:35]: `DeleteStale` would require reading all index documents and then collecting their UIDs before passing them to `DeleteDocumentsBatch`.
There's a feature that's not implemented yet: https://github.com/meilisearch/product/discussions/284
After it's in the Meilisearch codebase we could implement `DeleteStale` and release a major version update.

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**